src/Microsoft.Health.Dicom.SqlServer/Features/Store/SqlIndexDataStoreV1.cs

```diff
@@ -83,28 +83,28 @@ public virtual async Task<long> BeginCreateInstanceIndexAsync(Partition partitio
         }
     }
 
-    public virtual async Task<IEnumerable<VersionedInstanceIdentifier>> DeleteInstanceIndexAsync(Partition partition, string studyInstanceUid, string seriesInstanceUid, string sopInstanceUid, DateTimeOffset cleanupAfter, CancellationToken cancellationToken)
+    public virtual Task<IReadOnlyCollection<VersionedInstanceIdentifier>> DeleteInstanceIndexAsync(Partition partition, string studyInstanceUid, string seriesInstanceUid, string sopInstanceUid, DateTimeOffset cleanupAfter, CancellationToken cancellationToken)
```

Comment:
`DeleteInstanceIndexAsync`, `DeleteSeriesIndexAsync`, and `DeleteStudyIndexAsync` do not need to be `virtual`. Only the method `DeleteInstanceAsync`